ab_host-api/storage.adoc

```diff
@@ -3,11 +3,46 @@
 
 Interface for accessing the storage from within the runtime.
 
-IMPORTANT: As of now, the storage API should silently ignore any keys that start
-with the `:child_storage:default:` prefix. This applies to reading and writing.
-If the function expects a return value, then _None_ (<<defn-option-type>>)
-should be returned. See
-https://github.com/paritytech/substrate/issues/12461[substrate issue #12461].
+[#sect-storage-assumptions]
+==== Implementation Assumptions
+The storage and child storage (<<sect-child-storage-api>>) API in the Substrate
+implementation makes some behavioral assumptions on the underlying storage
+architecture. While Polkadot Host implementers can decide for themselves on how
+those APIs are implemented, those behaviors must be replicated in order for the
+Runtime to be executed deterministically. In Substrate, child storages are
```

Comment:
I don't understand the purpose of this paragraph. You seem to be literally describing what a specification is.

Reply:
So we've been telling the implementers that the "main" storage is fully separated from the child storages and that this functionality can be implemented however they want. But with this new specification we're basically saying that those storages are really not that separate, meaning we kind of insist now that child storage entries are namespaces by a specific prefix and that the Storage API actually _can_ access Child Storages. This paragraph is meant to imply that the Host implementation must simply deal with this "ugliness", but I guess I can clarify this further.

Regarding your other points.

> Why is this not explained under ext_storage_get_version_1?
> And why "for example"? We want a precise list, not examples?

I put everything into one section so it's clear how the underlying data is structured in Substrate implementation. Those things must also be considered for `read`, `append`, `next_key`, etc, too. But I should replace the "for example" there with something more formal.

> Similarly, why not explain this under ext_storage_clear_prefix?

As mentioned above, it's to keep things packed together; functions that modify data simply check whether the key starts with `:child_storage:`, but the `clear_prefix` function makes an exception by also checking whether the input is a substring of `:child_storage:`. If you think it's more appropriate to move that section to `ext_storage_clear_prefix` I can do so.